pymchelper/utils/dicom2sobp.py

```diff
@@ -0,0 +1,216 @@
+                LayerEnergy = NominalEnergy
+                LayerNumber = LayerNumber + 1
+
+            Layer = np.empty(NoPosition)
+            Layer.fill(LayerNumber)
+            data = np.insert(data, 0, Layer, axis=1)
+
+            Energy = np.empty(NoPosition)
+            Energy.fill(NominalEnergy)
+            data = np.insert(data, 1, Energy, axis=1)
+
+            weight = np.array(ds['IonBeamSequence'][0]['IonControlPointSequence'][i]['ScanSpotMetersetWeights'].value)
+            data = np.insert(data, 2, weight*Dose, axis=1)
+
+            if (i == 0):
+                CompleteData = data
+            else:
+                CompleteData = np.concatenate((CompleteData, data))
+
+        ZeroMUIndex = []
+        for j in range(len(CompleteData)):
+            if CompleteData[j, 2] == 0:
+                ZeroMUIndex.append(j)
+
+        CompleteData = np.delete(CompleteData, ZeroMUIndex, 0)
+
+        SpotIndex = np.arange(1, len(CompleteData)+1, step=1)
+        CompleteData = np.insert(CompleteData, 0, SpotIndex, axis=1)
+
+        self.nspots = len(CompleteData)
+        self.spots = [None] * self.nspots
+
+        for i, d in enumerate(CompleteData):
+            self.spots[i] = Spot(d[0], d[1], d[2], d[3], d[4], d[5])
+
+    def makesobp(self, bm):
+        """
+        """
+        pass
+
+    def writesobp(self, fn):
+        """
+        """
+        pass
+
+
+def main(args=None):
+    """ Main function of the dicom2sobp script.
+    """
+    if args is None:
+        args = sys.argv[1:]
+
+    # parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("plan", help="input treatmentplan")  # , type=argparse.FileType('r'))
+    parser.add_argument("-v", "--verbosity", action='count', help="increase output verbosity", default=0)
+    parser.add_argument("-f", "--flip-xy", action='store_true', help="flips x-y axis", dest="flipxy", default=False)
+    parser.add_argument("-b", "--beam-model", nargs='?', help="CSV beam model", type=argparse.FileType('r'),
+                        dest="bmodel")
+    parser.add_argument('-o', '--outfile', nargs='?', type=argparse.FileType('w'),
+                        help='output file, in SH12A sobp.dat format. Default: "sobp.dat"', default="SOBP_New_Plan.dat")
+
+    parsed_args = parser.parse_args(args)
+
+    if parsed_args.verbosity == 1:
+        logging.basicConfig(level=logging.INFO)
+    elif parsed_args.verbosity > 1:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig()
+
+    finp = parsed_args.plan
+    binp = parsed_args.bmodel
+    fout = parsed_args.outfile
+
+    plan = Plan(finp)
+    if binp:
+        bm = BeamModel(binp)
+
+    psum = 0.0
+    for spot in plan.spots:
+        en = spot.energy  # nominal energy
+
+        pn = bm.f_ppmu(en) * spot.mu * 1e6  # number of primaries
+        psum += pn
+
+        s = ""
+        s += "{:.5f}".format(bm.f_e(en) / 1000.0)  # actual energy [GeV]
```

Comment:
Consider using f-strings

```suggestion
        s += f"{bm.f_e(en) / 1000.0:.5f}"  # actual energy [GeV]
```